src/main/scala/dpla/ingestion3/mappers/providers/HarvardMapping.scala

```diff
@@ -0,0 +1,377 @@
+  }
+
+  override def originalRecord(data: Document[NodeSeq]): ExactlyOne[String] =
+    Utils.formatXml(data)
+  
+  override def isShownAt(data: Document[NodeSeq]): ZeroToMany[EdmWebResource] = {
+    val artMuseumLink = (data \ "metadata" \ "mods" \ "location" \ "url")
+      .flatMap(node => getByAttribute(node, "displayLabel", "Harvard Art Museums"))
+      .flatMap(node => getByAttribute(node, "access", "object in context"))
+      .flatMap(extractString(_))
+      .map(stringOnlyWebResource)
+
+    val collectionLinks =
+      collection(data)
+        .flatMap(_.title)
+        .flatMap(collectionTitle => {
+          (data \ "metadata" \ "mods" \ "location" \ "url")
+            .flatMap(node => getByAttribute(node, "displayLabel", collectionTitle))
+            .flatMap(node => getByAttribute(node, "access", "object in context"))
+            .flatMap(extractString(_))
+            .map(stringOnlyWebResource)
+      })
+
+    val objectInContext = (data \ "metadata" \ "mods" \ "location" \ "url")
+      .flatMap(node => getByAttribute(node, "displayLabel", "Harvard Digital Collections"))
+      .flatMap(node => getByAttribute(node, "access", "object in context"))
+      .flatMap(extractString(_))
+      .map(stringOnlyWebResource)
+
+    artMuseumLink ++ collectionLinks ++ objectInContext
+  }
+
+  //<mods:location><mods:url access="preview">
+  override def preview(data: Document[NodeSeq]): ZeroToMany[EdmWebResource] =
+    for {
+      node <- data \ "metadata" \ "mods" \ "location" \ "url"
+      if node \@ "access" == "preview"
+    } yield uriOnlyWebResource(URI(node.text.trim))
+
+  override def provider(data: Document[NodeSeq]): ExactlyOne[EdmAgent] = EdmAgent(
+    name = Some("Harvard Library"),
+    uri = Some(URI("http://dp.la/api/contributor/harvard"))
+  )
+
+  //utility
+
+  override def useProviderName: Boolean = false
+
+  override def getProviderName: String = "harvard"
+
+  override def originalId(implicit data: Document[NodeSeq]): ZeroToOne[String] =
+    extractString(data \\ "header" \ "identifier").map(_.trim)
+
+  override def sidecar(data: Document[NodeSeq]): JValue =
+    ("prehashId", buildProviderBaseId()(data)) ~ ("dplaId", mintDplaId(data))
+
+  override def dplaUri(data: Document[NodeSeq]): ZeroToOne[URI] = mintDplaItemUri(data)
+
+  private def name(name: Node): ZeroToOne[String] = name match {
+    case elem: Elem =>
+      val text = elem.text.trim
+      if (text.isEmpty) None
+      else Some(text)
+    case _ =>
+      None
+  }
+
+  private def processTitleInfo(titleInfo: Node): ZeroToOne[String] = titleInfo match {
+    case elem: Elem =>
+      val candidates = for {
+        child <- elem.child
+        if child.isInstanceOf[Elem]
+      } yield child.text.trim
+
+      if (candidates.isEmpty) None else Some(candidates.mkString(" ").trim)
+
+    case _ => None
+  }
+
+
+  //Helper method to get a list of creators and contributors
+  private def processNames(data: Document[NodeSeq]): Names = {
```

Comment:
I think of you made this a `val` instead of a `def` it would only execute once per mapping.  Probably not a big deal since Harvard is small.

Reply:
done